database/driver.mjs

```diff
@@ -161,3 +183,59 @@ class dbDriver {
 }
 
 export default dbDriver
+
+/**
+* Data belongs to or goes into different collections. The data 'type' usually tells us which one.
+* If no type is found on the data, use the provided override, if any.
+* 
+* @param data A data object or query object. The type will correspond to a collection
+* @param override If no type is on 'data', consider the provided override to be the type.
+* @return a known type string, such as "Project", or null
+*/ 
+function determineDataType(data, override) {
+ if (data._sub) return "User"
+ if (data.members) return "Group"
+ if (data.group) return "Project"
+ if (data.target && data.items) return "Page"
+ if (data.target && data.body) return "Line"
+ return override ?? data["@type"] ?? data.type
+}
+
+/**
+ * Determine the collection name based on data type.
+ * 
+ * @param type A type string, such as "Project", or null
+ * @return the corresponding collection name from environment variables
+ */
+function discernCollectionFromType(type) {
+  if (!type) return null
```

Comment:
I'm not sure if we want to be `null` here, but just `undefined` could mean a couple things, I guess.